packages/api/src/services/emp-stats.ts

```diff
@@ -0,0 +1,55 @@
+import assert from "assert";
+import * as uma from "@uma/sdk";
+export { BigNumber, utils } from "ethers";
+import { Currencies, AppState, PriceSample } from "..";
+import { calcTvl } from "../libs/utils";
+import Queries from "../libs/queries";
+type Config = {
+  currency?: Currencies;
+};
+type Dependencies = Pick<AppState, "emps" | "stats" | "prices" | "erc20s">;
+
+// this service is meant to calculate numbers derived from emp state, things like TVL, TVM and other things
+export default (config: Config, appState: Dependencies) => {
+  const { emps, stats, prices } = appState;
+  const { currency = "usd" } = config;
+
+  const queries = Queries(appState);
+  async function updateTvl(address: string) {
+    const emp = await queries.getAnyEmp(address);
+    // the full state has collateral decimals, pulled from erc20 state
+    const fullState = await queries.getFullEmpState(emp);
+
+    assert(uma.utils.exists(fullState.collateralCurrency), "Emp requires collateralCurrency: " + address);
+    const priceSample: PriceSample = prices[currency].latest[fullState.collateralCurrency];
+    assert(uma.utils.exists(priceSample), "No latest price found for emp: " + address);
+    const price = priceSample[1];
```

Comment:
why use the 1st index and not the 0th index? or is the index 1 the price and index 0 the timestamp? in either case a comment might be useful

Reply:
yea i will comment, the `PriceSample` type is `[timestamp,price]`